src/units/attack_type.hpp

```diff
@@ -135,15 +135,23 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 
 	// Configured as a bit field, in case that is useful.
 	enum AFFECTS { AFFECT_SELF=1, AFFECT_OTHER=2, AFFECT_EITHER=3 };
+	/**
+	 * Filter a list of abilities or weapon specials, removing any entries that don't own
+	 * the overwrite_specials attributes.
+	 *
+	 * @param overwriters list that may have overwrite_specials attributes.
+	 * @param priority value used below for checking of input list in overwrite_special_checking function.
+	 */
+	void overwrite_special_overwriter(unit_ability_list& overwriters, double& priority) const;
```

Comment:
If this will only be called within abilities.cpp, it can be a static function in that file, instead of needing to add it to the header file.